nimble_orchestration/components.py

```diff
@@ -168,3 +168,38 @@ def color(self):
         The color of this component.
         """
         return self._color
+
+
+class Shelf:
+    """
+    A class for all the orchestraion information relating to a shelf
+    """
+    def __init__(self,
+                 assembled_shelf: AssembledComponent,
+                 device: Device):
+        self._assembled_shelf = assembled_shelf
+        self._device = device
+
+    @property
+    def assembled_shelf(self):
+        """
+        Return the Object describing the assembled shelf (currently this in an empty
+        shelf in the correct location on the rack).
+        This is an AssembledComponent
+        """
+        return self._assembled_shelf
+
+    @property
+    def device(self):
+        """
+        Return the Device object for the networking component that sits on this shelf
+        """
+        return self._device
+
+    @property
+    def md(self):
+        """
+        Return the gitbuilding markdown for assembling this shelf
```

Comment:
```suggestion
        Return the GitBuilding markdown for assembling this shelf
```

Reply:
Also, is BuildUp the markdown format, or do you use that and GitBuilding interchangeably?

Reply:
The "flavour"/syntax is BuildUp. Here I mean "the markdown that is going to be processed by GitBuilding". I should be be more explicit